arduino-ide-extension/src/node/sketches-service-impl.ts

```diff
@@ -658,11 +665,59 @@ export class SketchesServiceImpl
     return FileUri.create(genBuildPath).toString();
   }
 
-  async getIdeTempFolderPath(sketch: Sketch): Promise<string> {
+  private async getIdeTempFolderPath(sketch: Sketch): Promise<string> {
     const sketchPath = FileUri.fsPath(sketch.uri);
     await fs.readdir(sketchPath); // Validates the sketch folder and rejects if not accessible.
     const suffix = crypto.createHash('md5').update(sketchPath).digest('hex');
-    return path.join(os.tmpdir(), `arduino-ide2-${suffix}`);
+    return path.join(
+      this.isTempSketch.tempDirRealpath,
+      `arduino-ide2-${suffix}`
+    );
+  }
+
+  async tempBuildPath(sketch: Sketch): Promise<string[]> {
+    const sketchPath = FileUri.fsPath(sketch.uri);
+    const { tempDirRealpath } = this.isTempSketch;
+    const tempBuildPaths = [
+      this.tempBuildPathMD5Hash(tempDirRealpath, sketchPath),
+    ];
+
+    // If on Windows, provide both the upper and the lowercase drive letter MD5 hashes. All together four paths are expected:
+    // One of them should match if the sketch is not yet compiled.
+    // https://github.com/arduino/arduino-ide/pull/1809#discussion_r1071031040
+    if (isWindows && Win32DriveRegex.test(tempDirRealpath)) {
+      const toggleFirstCharCasing = (s: string) =>
+        startsWithUpperCase(s) ? firstToLowerCase(s) : firstToUpperCase(s);
+      const otherCaseTempDirRealPath = toggleFirstCharCasing(tempDirRealpath);
+      tempBuildPaths.push(
+        this.tempBuildPathMD5Hash(otherCaseTempDirRealPath, sketchPath)
+      );
+      if (Win32DriveRegex.test(sketchPath)) {
+        const otherCaseSketchPath = toggleFirstCharCasing(sketchPath);
+        tempBuildPaths.push(
+          this.tempBuildPathMD5Hash(tempDirRealpath, otherCaseSketchPath),
+          this.tempBuildPathMD5Hash(
+            otherCaseTempDirRealPath,
+            otherCaseSketchPath
+          )
+        );
+      }
+    }
+    return tempBuildPaths;
+  }
+
+  private tempBuildPathMD5Hash(tempFolderPath: string, path: string): string {
+    return join(tempFolderPath, this.tempBuildFolderMD5Hash(path));
+  }
+
+  private tempBuildFolderMD5Hash(path: string): string {
+    const hash = crypto
+      .createHash('md5')
+      .update(path)
+      .digest('hex')
+      .toUpperCase();
+    const folderName = `arduino-sketch-${hash}`;
```

Comment:
This is correct for the version of Arduino CLI currently used by Arduino IDE, but it will break when the Arduino CLI dependency is bumped:

- https://github.com/arduino/arduino-cli/pull/2031
- https://github.com/arduino/arduino-cli/pull/2033

After those changes, the path is now `<temp dir>/arduino/sketches/<hash>`.

No action needed now, but I thought to add a note in anticipation of the adjustment that will be needed at the time of the Arduino CLI dependency bump.

Reply:
Thank you for the heads-up!

Reply:
I pushed a test for the build path: https://github.com/arduino/arduino-ide/pull/1809/commits/fd38ba494df6ba5f75dca3579838519417e5a0e9.

I had to change the IDE2 logic when stopping and restarting the daemon. I have to double-check it.

Reply:
> I have to double-check it.

The `ready` deferred promise is not used anywhere else; [the change](https://github.com/arduino/arduino-ide/pull/1809/commits/fd38ba494df6ba5f75dca3579838519417e5a0e9#diff-f463d1c276a9080ec522ead62c6dd596e5bf330b093b7d51b71738c9795d0fdaL139-R139) must not break anything. We need to clean up this code next time we change the core gRPC clients.

Reply:
I have decided to drop everything I changed after the review and moved it to a separate PR: https://github.com/arduino/arduino-ide/pull/1823. If the build is green, I would like to merge it as it is if you are OK with it, Per. Thanks!

Reply:
Yes, I'm OK with that. Feel free to merge anytime you are ready Akos.